src/mmap_serialization_sink.rs

```diff
@@ -0,0 +1,18 @@
+use crate::serialization::{Addr, SerializationSink};
+use std::path::Path;
+
+pub struct MmapSerializationSink {}
```

Comment:
It might be worth implementing a plain sink on top of regular files first and then seeing what the performance difference is from using `mmap`'d files. Just an idea though.